channel.js

```diff
@@ -559,6 +559,7 @@ TChannel.prototype.listen = function listen(port, host, callback) {
     assert(typeof host === 'string', 'TChannel requires host argument');
     assert(typeof port === 'number', 'TChannel must listen with numeric port');
     assert(host !== '0.0.0.0', 'TChannel must listen with externally visible host');
+    assert(!self.topChannel, 'TChannel must listen on top channel');
```

Comment:
This is the opposite assertion of what you want:
- chan.topChannel is not a bool
- chan.topChannel is a reference to the top channel, if chan is a sub channel